// @rsbuild/core@0.3.2/dist/provider/core/initHooks.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var initHooks_exports = {};
__export(initHooks_exports, {
  initHooks: () => initHooks
});
module.exports = __toCommonJS(initHooks_exports);
var import_shared = require("@rsbuild/shared");
function initHooks() {
  return {
    /** parameters are not bundler-related */
    onExit: (0, import_shared.createAsyncHook)(),
    onDevCompileDone: (0, import_shared.createAsyncHook)(),
    onAfterStartDevServer: (0, import_shared.createAsyncHook)(),
    onBeforeStartDevServer: (0, import_shared.createAsyncHook)(),
    onAfterStartProdServer: (0, import_shared.createAsyncHook)(),
    onBeforeStartProdServer: (0, import_shared.createAsyncHook)(),
    /** parameters are bundler-related */
    onAfterBuild: (0, import_shared.createAsyncHook)(),
    onBeforeBuild: (0, import_shared.createAsyncHook)(),
    modifyRspackConfig: (0, import_shared.createAsyncHook)(),
    modifyBundlerChain: (0, import_shared.createAsyncHook)(),
    modifyWebpackChain: (0, import_shared.createAsyncHook)(),
    modifyWebpackConfig: (0, import_shared.createAsyncHook)(),
    modifyRsbuildConfig: (0, import_shared.createAsyncHook)(),
    onAfterCreateCompiler: (0, import_shared.createAsyncHook)(),
    onBeforeCreateCompiler: (0, import_shared.createAsyncHook)()
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  initHooks
});
